rest-api/authentication/routes/auth-routes.spec.js

```diff
@@ -0,0 +1,105 @@
+const request = require('supertest');
+const server = require('../../../api/server');
+
+function random(option) {
+    const chars = 'abcdefghijklmnopqrstuvwxyz1234567890';
+    let string = '';
+    for (var i = 0; i < 15; i++) {
+        string += chars[Math.floor(Math.random() * chars.length)];
+    }
+    if (option === 'email') {
+        return string + '@email.com'
+    } else {
+        return string;
+    }
+}
+
+const register = {
+    email: random('email'),
+    username: random(),
+    password: 'password'
+}
+
+const unregistredAccount = {
+    email: random('email'),
+    username: random(),
+    password: 'password'
+}
+
+
+// Add Register
+
+describe('POST /api/auth/register', () => {
+    it('Create a new User and return ', () => {
+        return request(server)
+            .post('/api/auth/register')
```

Comment:
Hmm... reason for preference over async/await, here? :)

Reply:
*and elsewhere?

Reply:
I think it looks a lot cleaner in this case